src/clients/email/email.client.ts

```diff
@@ -0,0 +1,68 @@
+/* eslint-disable camelcase */
+import { SESService } from '@services/ses.service';
+import nodemailer, { SentMessageInfo, Transporter } from 'nodemailer';
+import { USE_AWS_SES, SENDGRID_API_KEY, SENDGRID_API_USER, USE_SENDGRID } from '@config';
+import * as sgTransport from 'nodemailer-sendgrid-transport';
+import aws from 'aws-sdk';
+import SESTransport from 'nodemailer/lib/ses-transport';
+import { EmailInterface } from '@interfaces';
+import SMTPTransport from 'nodemailer/lib/smtp-transport';
+
+export class EmailClient {
+  transporter: Transporter<SentMessageInfo>;
+  private static instance: EmailClient;
+
+  public static getInstance(): EmailClient {
+    if (!this.instance) {
+      this.instance = new EmailClient();
+    }
+
+    return this.instance;
+  }
+
+  constructor() {
+    if (USE_AWS_SES() && USE_SENDGRID()) {
```

Comment:
I dont understand this line, can you explain me why the parameters are invoked as functions?

Reply:
Fixed, thank you !